File: backend/src/main/java/cohappy/backend/model/dto/response/DebtDTO.java
package cohappy.backend.model.dto.response;

import cohappy.backend.model.DebtType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DebtDTO {
    @Id
    private String debtId;
    private String creditorUserCode;
    private Map<String,Boolean> debtorsUserCode;
    private boolean isCreatorIncluded;
    private float amount;
    private String description;
    private DebtType debtType;
}

File: frontend/app/src/main/java/cohappy/frontend/client/dto/response/DebtDTO.kt
package cohappy.frontend.client.dto.response
import cohappy.frontend.client.dto.enum.DebtType
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class DebtDTO(
    var debtId: String? = null,
    var linkedDebtId: String? = null,
    var creditorUserCode: String? = null,
    var debtorsUserCode: Map<String,Boolean>? = null,
    var isCreatorIncluded: Boolean? = null,
    var amount: Float? = null,
    var description: String? = null,
    val debtType: DebtType
)